src/Domain/CoordinateMaps/DiscreteRotation.cpp

```diff
@@ -16,7 +16,20 @@ namespace CoordinateMaps {
 template <size_t VolumeDim>
 DiscreteRotation<VolumeDim>::DiscreteRotation(
     OrientationMap<VolumeDim> orientation) noexcept
-    : orientation_(std::move(orientation)) {}
+    : orientation_(std::move(orientation)) {
+  ASSERT(
+      [&orientation]() {
+        std::set<size_t> set_of_dims;
+        for (size_t j = 0; j < VolumeDim; j++) {
+          set_of_dims.insert(
+              orientation(gsl::at(Direction<VolumeDim>::all_directions(), j))
+                  .dimension());
+        }
+        return set_of_dims.size() == VolumeDim;
+      }(),
+      "OrientationMap is invalid! "
+      "You might be indexing into a vector of OrientationMaps incorrectly.");
```

Comment:
Could you print some info for the user to be able to external diagnose the issue without having to add more print statements in here?